content/en/blog/_posts/2023-01-15-Security-Bahavior-Analysis/index.md

```diff
@@ -0,0 +1,87 @@
+---
+layout: blog
+title: "All Microservices Are Vulnerable"
+date: 2023-01-15
+slug: security-behavior-analysis
+---
+
+**Author:**
+David Hadas (IBM Research Labs)
+
+## Avoiding a False Sense of Security
+
+_This post warns Devops from a false sense of security. Following security best practices when developing and configuring microservices do not result in non-vulnerable microservices. The post shows that although all deployed microservices are vulnerable, there is much to be done to ensure microservices are not exploited. It explains how analyzing the behavior of clients and services from a security standpoint, named here **"Security-Behavior Analysis"**, can protect the deployed vulnerable microservices. It points to [Guard](http://knative.dev/security-guard), an open source project offering security-behavior monitoring and control of Kubernetes microservices presumed vulnerable._
+
+As cyber war games continue to intensify in sophistication, organizations deploying cloud services, continue to grow their cyber investments aiming to produce safe and non-vulnerable services. However, the year-by-year growth in cyber investments, does not result in a parallel reduction in cyber incidents. Instead, the number of cyber incidents continues to grow annually. Evidently organizations are fighting a battle they are doomed to fail at. No matter how much effort is made to detect and remove cyber weaknesses from deployed services, it seems offenders have the upper hand.
+
+Considering the current spread of offensive tools, sophistication of offensive players, and ever-growing cyber financial gains to offenders, any cyber strategy that relies on constructing a non-vulnerable, weakness-free service in 2023 is clearly too naïve. It seems the only viable strategy is to:
+
+&#x27A5; **Admit that your services are vulnerable!**
+
+In other words, consciously accept that you will lose any battle aimed to create non-vulnerable services. If your opponents find a single weakness as an entry-point, you lose. Admitting that in spite of your best efforts, all your services are most likely vulnerable is an important first step. Next, we need to discuss what can you do about it...
```

Comment:
Can we possibly rephrase this to the below? I don't think it takes away from the overall meaning. WDYT?
```suggestion
Consciously admitting that in spite of your best efforts, all your services are most likely vulnerable is an important first step. Next, we need to discuss what can you do about it...
```

Reply:
I will rephrase without the word "battle".